airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import re
+from abc import abstractmethod
+from typing import Any, Dict, List
+
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
+from jsonschema import RefResolver
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class AbstractFileBasedSpec(BaseModel):
+    """
+    Used during spec; allows the developer to configure the cloud provider specific options
+    that are needed when users configure a file-based source.
+    """
+
+    streams: List[FileBasedStreamConfig] = Field(
+        title="The list of streams to sync",
+        description="Streams defines the behavior for grouping files together that will be synced to the downstream destination. Each "
+        "stream has it own independent configuration to handle which files to sync, how files should be parsed, and the "
+        "validation of records against the schema.",
+        order=10,
+    )
+
+    @classmethod
+    @abstractmethod
+    def documentation_url(cls) -> AnyUrl:
+        """
+        :return: link to docs page for this source e.g. "https://docs.airbyte.com/integrations/sources/s3"
+        """
+
+    @classmethod
+    def schema(cls, *args: Any, **kwargs: Any) -> Dict[str, Any]:
+        """
+        Generates the mapping comprised of the config fields
+        """
+        schema = super().schema(*args, **kwargs)
+        resolved_schema = cls.resolve_refs(schema)
+        resolved_schema = cls.add_legacy_format(resolved_schema)
+
+        return resolved_schema
+
+    @staticmethod
+    def resolve_refs(schema: dict) -> dict:
```

Comment:
ah okay yeah i didn't know we were reusing. I think that would make sense since this satisfies the common use case of pydantic models being used as source of a spec being generated. I'll add this to our common utils

Reply:
scratch that, TIL we actually already have a utility for that here, so we can releverage what we already have https://github.com/airbytehq/airbyte/blob/2e099acc52aeddc172c5ee66ed98426c69449a4a/airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py#L79-L85